needle/modules/storage/data/keychain_dump.py

```diff
@@ -32,23 +35,54 @@ def module_pre(self):
     # ==================================================================================================================
     def module_run(self):
         # Composing the command string
-        cmd = '{}'.format(self.device.DEVICE_TOOLS['KEYCHAINDUMPER'])
+        cmd = '{}'.format(self.device.DEVICE_TOOLS['KEYCHAIN_DUMP'])
         msg = "Dumping the keychain"
         if self.options['filter']:
-            cmd += ' | grep -i "{}" -A 3 -B2'.format(self.options['filter'])
+            grep_args = ' -i -a "{}" -C 10 '.format(self.options['filter'])

```

Comment:
yes, but you are applying the filter later on (line 64). 
Keychain_dump create files, there is no need to grep directly when invoking it.